File: OptiPlanBackend/OptiPlanBackend/Models/Team.cs
namespace OptiPlanBackend.Models
{
    public class Team
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;

        public Guid ProjectId { get; set; }
        public Project Project { get; set; }

        public ICollection<TeamMembership> Members { get; set; } = new List<TeamMembership>();
        public ICollection<Invitation> Invitations { get; set; } = new List<Invitation>();
    }
}

File: optiplan-frontend/src/app/models/team.ts
import { Invitation } from "./invitation";
import { Project } from "./project";
import { TeamMembership } from "./team-membership";

export interface Team {
  id: string;
  name: string;
  projectId: string;
  project: Project;
  members: TeamMembership[];
  invitations: Invitation[];
  createdAt: string; // ISO format, à convertir si tu veux une Date avec new Date(createdAt)
}
